### PR TITLE
Publisher jms output adapter

### PR DESCRIPTION
**Pull request by keizer619**

### CODE REVIEW

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java [hunk 1]
```diff
@@ -247,8 +250,12 @@ public void run() {
                 this.processAuthentication(method);
                 this.processHeaders(method);
                 httpClient.execute(method).getEntity().getContent().close();
-            } catch (Exception e) {
-                log.error("Error executing HTTP output event adapter sender: " + e.getMessage(), e);
+            }
+            catch (UnknownHostException e){
+                throw new ConnectionUnavailableException("Exception while connecting HTTP endpoint ", e);
+            }
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java]: Add context

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java [hunk 2]
```diff
@@ -247,8 +250,12 @@ public void run() {
...
+            catch (Exception e) {
+                throw new OutputEventAdapterRuntimeException("Error executing HTTP output event adapter sender: ", e);
             }
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java]: Add context

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/pom.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    <description>org.wso2.carbon.event.output.adapter.jms provides the back-end functionality of jms event adaptor
+    </description>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.carbon.event-processing</groupId>
+            <artifactId>org.wso2.carbon.event.output.adapter.core</artifactId>
+            <version>2.0.4-SNAPSHOT</version>
+        </dependency>
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/pom.xml]: remove the version from here

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            publisherDetails.getJmsMessageSender().send(jmsMessage, publisherDetails.getMessageConfig());
+    }
+
+    @Override
+    public void disconnect() {
+        if (publisherDetails != null) {
+                publisherDetails.getJmsMessageSender().close();
+                publisherDetails.getJmsConnectionFactory().stop();
+            }
+    }
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java]: Format the code

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            String[] headers = headerProperty.split(",");
+
+            if (headers != null && headers.length > 0) {
+                for (String header : headers) {
+                    String[] headerPropertyWithValue = header.split(":");
+                    if (headerPropertyWithValue.length == 2) {
+                        messageConfiguration.put(headerPropertyWithValue[0], headerPropertyWithValue[1]);
+                    } else {
+                        log.warn("Header property not defined in the correct format");
+                    }
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java]: add context : print headerPropertyWithValue

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.soap/src/main/java/org/wso2/carbon/event/output/adapter/soap/SoapEventAdapter.java [hunk 6]
```diff
@@ -208,9 +208,8 @@ public void run() {
                     serviceClient.setOptions(options);
                     serviceClient.fireAndForget(AXIOMUtil.stringToOM(payload.toString()));
 
-                } catch (AxisFault axisFault) {
-                    throw new OutputEventAdapterRuntimeException(
-                            "Exception while sending events to soap endpoint ", axisFault);
+                } catch (AxisFault e) {
+                    throw new ConnectionUnavailableException("Exception while sending events to soap endpoint ", e);
                 } catch (XMLStreamException e) {
```
Comment (suhothayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.soap/src/main/java/org/wso2/carbon/event/output/adapter/soap/SoapEventAdapter.java]: Add context to the exception

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/pom.xml (hunk 3 above)
Comment (keizer619) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/pom.xml]: Fixed

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java (hunk 4 above)
Comment (keizer619) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java]: Fixed

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java (hunk 5 above)
Comment (keizer619) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.jms/src/main/java/org/wso2/carbon/event/output/adapter/jms/JMSEventAdapter.java]: Context added

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java (hunk 1 above)
Comment (keizer619) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java]: Context Added

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.soap/src/main/java/org/wso2/carbon/event/output/adapter/soap/SoapEventAdapter.java (hunk 6 above)
Comment (keizer619) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.soap/src/main/java/org/wso2/carbon/event/output/adapter/soap/SoapEventAdapter.java]: Context Added

**Pull request merged**